src/watchmaker/workers/salt.py

```diff
@@ -661,6 +667,49 @@ def _set_grain(self, grain, value):
         self.log.info('Setting grain `%s` ...', grain)
         super(SaltLinux, self)._set_grain(grain, value)
 
+    def _install_pip_modules(self):
+        if self.pip_install:
+            python = self.run_salt(['grains.get', 'pythonexecutable'])
+            py_exec = python['stdout'].decode().split('\n')[1].strip()
+            self.log.info('Salt Python interpreter: `%s`', py_exec)
+
+            try:
+                ensure = self.call_process([py_exec, '-m', 'ensurepip'])
+                self.log.info('Ensure pip: %s', ensure['stdout'])
+                ver = self.call_process([py_exec, '-m', 'pip', '--version'])
+                self.log.info('Pip version: %s', ver['stdout'])
+            except WatchmakerException:
+                self.log.info('Pip not installed for Salt interpreter!')
+                self.log.info('Attempting pip install...')
+                get_pip = [
+                    'curl',
+                    'https://bootstrap.pypa.io/get-pip.py',
+                    '-o',
+                    'get-pip.py'
+                ]
+                self.call_process(get_pip)
+                self.call_process([py_exec, 'get-pip.py'])
+
+            for module in self.pip_install:
+                self.log.info('Pip installing module: `%s`', module)
+                pip_cmd = [
+                    py_exec,
+                    '-m',
+                    'pip',
+                    'install',
+                    module,
+                    '--ignore-installed'
```

Comment:
Why add this option? Also, systems that do not have connectivity to the public pypi will need to be able to pass their own options, at least to set the pypi index server. Probably need to consider another config value for user-specified pip options.

Reply:
The system interpreter has pure distutils installed packages that make pip fail when installing. For example, requests is distutils installed and is a prerequisite of hvac so it fails without this option.